### PR TITLE
[mempool] Add counter for created block size in txn count and byte size

### PR DESCRIPTION
**Pull request by igor-aptos**
We have txn/committed block counter, but without re-serializing things there, I cannot easily get the size. This seems an easy place to measure it, and it also enables us to easily see if different nodes are trying to propose different sizes as well (irrespective of whether those land on the blockchain)

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4460)

### CODE REVIEW

**Review by sitalkedia — APPROVED** [review]
LGTM

File: mempool/src/core_mempool/mempool.rs [hunk 1]
```diff
@@ -201,6 +201,9 @@ impl Mempool {
             block_size = block.len(),
             byte_size = total_bytes,
         );
+
+        MEMPOOL_RETURNED_NUM_TXNS_PER_BLOCK.observe(block.len() as f64);
```
Comment (zekun000) [inline mempool/src/core_mempool/mempool.rs:205]: I thought we already have a counter for this

**Review by zekun000 — APPROVED** [review]

**Comment by igor-aptos** [comment]
we actually did have a counter for txns.len() already - mempool_service_transactions

deduping that, and moving where that is logged, so that it is logged at the same place as byte size.

@bchocho for if this is better (as this is within mempool)

**Review by bchocho — APPROVED** [review]
LGTM

**Pull request merged**